File: 3005-count-elements-with-maximum-frequency/3005-count-elements-with-maximum-frequency.js
var maxFrequencyElements = function(nums) {
    const map = new Map();

    for (const num of nums) {
        map.set(num, (map.get(num) || 0) + 1);
    }

    let maxFreq = 0;
    for (const freq of map.values()) {
        maxFreq = Math.max(maxFreq, freq);
    }

    let sum = 0;
    for (const freq of map.values()) {
        if (freq === maxFreq) {
            sum += maxFreq;
        }
    }

    return sum;
};

File: 3005-count-elements-with-maximum-frequency/3005-count-elements-with-maximum-frequency.java
class Solution {
    public int maxFrequencyElements(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        
        for(int num: nums){
            map.put(num, map.getOrDefault(num, 0)+1);
        }
        int maxfreq =0;
        for(int freq: map.values()){
            maxfreq = Math.max(maxfreq, freq);

        }
        int sum =0;
        for(int freq: map.values()){
            if(freq == maxfreq){
                sum+=maxfreq;
            }
        }
        return sum;

    }
}